rpcs/ledgerService.go

```diff
@@ -245,12 +274,15 @@ func (ls *LedgerService) sendCatchupRes(ctx context.Context, target network.Unic
 }
 
 func (ls *LedgerService) encodedBlockCert(round uint64) ([]byte, error) {
-	blk, cert, err := ls.ledger.BlockCert(basics.Round(round))
+	blk, cert, err := ls.ledger.EncodedBlockCert(basics.Round(round))
 	if err != nil {
 		return nil, err
 	}
-	return protocol.Encode(EncodedBlockCert{
-		Block:       blk,
-		Certificate: cert,
-	}), nil
+	outBuffer := make([]byte, len(blk)+len(cert)+len(ls.encodingWrappers[0])+len(ls.encodingWrappers[1])+len(ls.encodingWrappers[2]))
```

Comment:
this works because msgpack effectively says:
`"map of length 2" {key} {value} {key} {value}`
and so we can drop in any valid msgpack object for those `{value}` parts.
The alternative is that there appears to be a go-codec.Raw type which is an alias for []byte which gets passed through and copied into the output. I think it was designed for this very purpose. We could make a `type EncodedBlockCertPreencoded struct {...}` that duplicated the fields except had them as Raw. I'm not sure that's actually better and I haven't tested that it would work.

Reply:
Directly using codec.Raw triggers a panic "Raw values cannot be encoded"; I'm sure it's possible to work around that, but I haven't spend enough time figuring out how to. I think that setting the EncodeOptions.Raw = true would work around that.

Reply:
Indeed, I think setting `Raw = true` in our codec params should do it.  Here's a sketch that seems to work:

https://github.com/zeldovich/go-algorand/commit/683816aef98230ded2d04d476708628edc39854b

Reply:
Yes, your solution aligns with Brian's and is more clean. I've merged it into this PR.